pinot-core/src/main/java/org/apache/pinot/core/upsert/UpsertMetadataTableManager.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.upsert;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.annotation.concurrent.ThreadSafe;
+import org.apache.pinot.core.data.manager.realtime.RealtimeTableDataManager;
+import org.apache.pinot.core.realtime.impl.ThreadSafeMutableRoaringBitmap;
+import org.apache.pinot.spi.data.readers.PrimaryKey;
+
+/**
+ * The manager of the upsert metadata of a table.
+ */
+@ThreadSafe
+public class UpsertMetadataTableManager {
```

Comment:
I went through the code, and seems we don't need to do the table level management, partition level management should be good enough. Directly using `PartitionUpsertMetadataManager` can save lots of unnecessary map lookups, and eliminate the contention of the concurrent lookups

Reply:
I'd like to keep this class for (1) a group of the `PartitionUpsertMetadataManager ` (note `PartitionUpsertMetadataManager ` is package-private); (2) more features might be added to this class, such as future offline table support, record retrieval for partial update, etc

Reply:
Also why would there be concurrent lookup contention, since we don't the synchronized keywords on most methods?

Reply:
There won't be lots of partitions, and all the lookups are very likely hitting the same partition, which can cause contention for concurrent hash map. It is okay to keep this class for easier management, but try to directly use partition level manager instead of look up this one

Reply:
`try to directly use partition level manager instead of look up this one` can you be more specific on this? The current methods in this class delegate to the partition level manager?